problems/[230608] Number of Islands.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * leetcode problem link: https://leetcode.com/problems/number-of-islands
+ *
+ * @param {character[][]} grid
+ * @return {number}
+ */
+const numIslands = function (grid) {
+  let islandCount = 0;
```

Comment:
```suggestion
  return grid.reduce((acc, row, rowIdx) => {
    return acc + row.reduce((acc, col, colIdx) => acc + findCliff(rowIdx, colIdx), 0);
  }, 0);  
```

물론 함수 내에서 가변 변수를 얼마든지 사용할 수 있지만, 혹시 모를 변수를 줄이고자 한다면 reduce를 활용해볼 수도 있을 것 같습니다..!